### ISSUE TITLE
How to know it is the training phase or the inference phase in operators in new framework?

### ISSUE THREAD

**Issue by qingqing01** [PaddlePaddle/Paddle#4108]
Some operator‘s calculation, such `Dropout`, `BatchNorm` is different between training phase and testing phase. Now,  the `run` function of the operator only needs `core.Scope` and `core.DeviceContext`, and how to run operator in Python API as follows:

```python
net_op = core.Net.create()
# create fc_op
# net_op.append_op(fc_op)
# net_op.append_op(mse_op)
ctx = core.DeviceContext.create(core.CPUPlace())
scope = core.Scope()
# scope.new_var('X')
# ...
net_op.run(scope, ctx)
```

The op can not determine whether it is training phase or inference phase. And it's strange to put this flag into `DeviceContext`. And the `ExecutionContext` is not exposed to users in Python API. 

This problem had been discussed in the `Hi` discussion group.  Now the dropout operator does not handle this case.  I just record this problem here.
